### DISCUSSION TITLE
Reset environment in the Python code?

### DISCUSSION THREAD

**Discussion by criddle858** [QuirkyCort/gears#108] [Ideas]
Is there a way to reset the environment (robot, all objects, etc.) in the Python code? I know you can reset manually. I want to reset the environment and run code in a loop (for ML).

**Comment by QuirkyCort** [comment]
Sorry for taking so long. Been busy lately.

It wasn't possible before, but I've added it in. Use it like this...

```
from simPython import System

sys = System()

while True:
    motorD = LargeMotor(OUTPUT_D) # Arm
    motorE = LargeMotor(OUTPUT_E) # Paintball Launcher

    motorD.on_for_degrees(5, 43)
    motorE.on_for_degrees(100, (-1000))
    motorE.on_for_degrees(100, 1)
    time.sleep(3)
    sys.reset_simulator()
    time.sleep(1)
````

Few things to note...

* You need to create a new instance of every sensor / actuator after reset. This is because the reset destroys the original motors and sensors, making the original instances invalid.
* There's a short sleep after the reset. This is because sys.reset_simulator() may return before the reset is complete.

**Comment by QuirkyCort** [comment]
If you can share how you are using Gears for ML, that would be appreciated. Always keen to see new ways of using Gears.
